### ISSUE TITLE
Incorrect batch size info with keras.utils.Sequence

### ISSUE THREAD

**Issue by sciann** [tensorflow/tensorflow#50048]
I had reported this issue before, but I now find the source is in the `train_generator_v1.py`. 

TensorFlow 2.4/2.5

Batch size info is miscalculated when data is wrapped in Sequence. The example below is the output for the first epoch training on 1,000 points with batch_size=32, which should result in 32 mini-batch updates. The batch number on the far left is correct (32/32), however, the size and batch before loss seem to be incorrect.

Output:
Epoch 1/100
Total samples: 1000 
Batch size: 32 
Total batches: 32 

Epoch 1/10
32/32 [==============================] - 1s 2ms/step - **batch: 15.5000 - size: 31.2500** - loss: 0.1369
Epoch 2/10
32/32 [==============================] - 0s 1ms/step - **batch: 15.5000 - size: 31.2500** - loss: 0.0307
Epoch 3/10
32/32 [==============================] - 0s 1ms/step - **batch: 15.5000 - size: 31.2500** - loss: 0.0148

```python
import tensorflow as tf
import numpy as np 
print(tf.__version__)
from tensorflow import keras as k
tf.compat.v1.disable_eager_execution()

class Data(k.utils.Sequence):
    """
    Converts fit() into fit_generator() interface.
    """

    def __init__(self, inputs, outputs, sample_weights, batch_size, shuffle):
        self._inputs = inputs
        self._outputs = outputs
        self._sample_weights = sample_weights
        self._size = inputs[0].shape[0]
        self._batch_size = batch_size
        self._num_batches = int((self._size-1)/batch_size) + 1
        self._shuffle = shuffle
        self._ids = np.arange(0, self._size)
        self._reshuffle()
        print("\nTotal samples: {} ".format(self._size))
        print("Batch size: {} ".format(min(self._batch_size, self._size)))
        print("Total batches: {} \n".format(self._num_batches))

    def __len__(self):
        return self._num_batches

    def __getitem__(self, index):
        start = index * self._batch_size
        end = min(start + self._batch_size, self._size)
        ids = self._ids[start: end]
        inputs = [v[ids, :] for v in self._inputs]
        outputs = [v[ids, :] for v in self._outputs]
        sample_weights = [v[ids] for v in self._sample_weights]
        return inputs, outputs, sample_weights
    
    def on_epoch_end(self):
        self._reshuffle()

    def get_data(self):
        return self._inputs, self._outputs, self._sample_weights

    def _reshuffle(self):
        if self._num_batches > 1 and self._shuffle:
            self._ids = np.random.choice(self._size, self._size, replace=False)


x = k.Input((1,))
l1 = k.layers.Dense(10, activation='tanh')(x)
y = k.layers.Dense(1)(l1)

model = k.Model(x, y)
model.compile(loss=k.losses.MSE)

inputs = [np.linspace(0, 1, 1000).reshape(-1,1)]
outputs = list(map(lambda x: np.sin(2*x), inputs))
weights = list(map(lambda x: np.ones(x.size), inputs))

dg = Data(inputs, outputs, weights, 32, True)

model.fit(dg, epochs=10)
````

**Comment by UsharaniPagadala** [comment]
@Saduf2019 
I was able to replicate the issue reported here.Please find the [gist](https://colab.research.google.com/gist/UsharaniPagadala/3d303f6652cfe77aebb37baf40201858/untitled78.ipynb).Thanks

**Comment by Saduf2019** [comment]
@sciann 

Please post this issue on [keras-team/keras repo](https://github.com/keras-team/keras/issues).
To know more see;
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
